library/src/compute/calc.rs

```diff
@@ -404,6 +408,170 @@ pub fn log(
     Value::Float(result)
 }
 
+/// Calculates the product of a range of numbers. Used to calculate permutations.
+/// Returns None if the result is larger than `u64::MAX`
+fn factorial_range(start: u64, end: u64) -> Option<u64> {
+    // By convention
+    if end + 1 < start {
+        return Some(0);
+    }
+
+    let mut count: u64 = 1;
+    let real_start: u64 = cmp::max(1, start);
+
+    for i in real_start..=end {
+        match count.checked_mul(i) {
```

Comment:
this could be either `count = count.checked_mul(i)` or the whole loop can be replaced with `try_fold`

Reply:
It wouldn't result in the same thing, with the `try_fold` it would return the latest value which is not an overflow instead of returning `None`. Also it is not directly possible to set `count = count.checked_mul(...)` becauses it makes `count` an option, which is not intended (and you can't call `checked_mul(..)` on an option)

Reply:
Sorry, I meant `count = count.checked_mul(i)?`. I forgot to type the `?`.